### PR TITLE
super-productivity: Add version 6.1.2

### LINKED ISSUE #5316: [Request] Super Productivity
Open source productivity app.

"(...) Super Productivity is a ToDo List / Time Tracker / Personal Jira Task Manager for Linux, MacOS and Windows (...)"

GitHub repo: https://github.com/johannesjo/super-productivity
GitHub realeases (binaries): https://github.com/johannesjo/super-productivity/releases

<a href="https://youtu.be/_mvMXWLsL-8">Work View with global links</a>

### PR DESCRIPTION
**Pull request by joao-neves95**
- Closes #5316

### CODE REVIEW

File: bucket/super-productivity.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "hash": "bc7523cffb1e9d64ae7ab0760c7c4432b743cfa5503c386de2a01f7da554c664",
+  "installer": {
+    "script": [
+        "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-64.7z\" \"$dir\"",
+        "Remove-Item \"$dir\\`$PLUGINSDIR\" -Force -Recurse"
+    ]
+  },
+  "uninstaller": {
+    "file": "Uninstall superProductivity.exe"
+  },
```
Comment (Ash258) [inline bucket/super-productivity.json:16]:
```suggestion
```

File: bucket/super-productivity.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "homepage": "https://super-productivity.com/",
+  "license": "MIT",
+  "url": "https://github.com/johannesjo/super-productivity/releases/download/v6.1.2/superProductivity-Setup-6.1.2.exe#/dl.7z",
+  "hash": "bc7523cffb1e9d64ae7ab0760c7c4432b743cfa5503c386de2a01f7da554c664",
+  "installer": {
+    "script": [
+        "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-64.7z\" \"$dir\"",
+        "Remove-Item \"$dir\\`$PLUGINSDIR\" -Force -Recurse"
+    ]
+  },
```
Comment (Ash258) [inline bucket/super-productivity.json:13]: 64bit only

File: bucket/super-productivity.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      [
+          "superProductivity.exe",
+          "superProductivity"
+      ]
+  ],
+  "checkver": {
+      "github": "https://github.com/johannesjo/super-productivity/releases"
+  },
+  "autoupdate": {
+      "url": "https://github.com/johannesjo/super-productivity/releases/download/v$version/superProductivity-Setup-$version.exe#/dl.7z"
```
Comment (Ash258) [inline bucket/super-productivity.json:28]: 64bit only
Add hash extraction from latest.yml

**Review by Ash258 — CHANGES_REQUESTED** [review]
Format manifest using checkver/formatjson

File: bucket/super-productivity.json (hunk 2 above)
Comment (joao-neves95) [reply bucket/super-productivity.json]: I've fixed the installer and added 64bit architecture only.
What do you think?

File: bucket/super-productivity.json (hunk 1 above)
Comment (joao-neves95) [reply bucket/super-productivity.json]: I have formatted the JSON doc.

File: bucket/super-productivity.json [hunk 4]
```diff
@@ -1,30 +1,39 @@
...
-  },
-  "autoupdate": {
-      "url": "https://github.com/johannesjo/super-productivity/releases/download/v$version/superProductivity-Setup-$version.exe#/dl.7z"
-  }
+    "version": "6.1.2",
+    "description": "To-do list & time tracker for programmers and other digital workers with Jira, Github, and Gitlab integration.",
+    "homepage": "https://super-productivity.com/",
+    "license": "MIT",
+    "url": "https://github.com/johannesjo/super-productivity/releases/download/v6.1.2/superProductivity-Setup-6.1.2.exe#/dl.7z",
+    "hash": "bc7523cffb1e9d64ae7ab0760c7c4432b743cfa5503c386de2a01f7da554c664",
```
Comment (Ash258) [inline bucket/super-productivity.json:7]: 64bit

File: bucket/super-productivity.json [hunk 5]
```diff
@@ -1,30 +1,39 @@
...
+                "script": [
+                    "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-64.7z\" \"$dir\"",
+                    "Remove-Item \"$dir\\`$PLUGINSDIR\" -Force -Recurse"
+                ]
+            }
+        }
+    },
+    "uninstaller": {
+        "file": "Uninstall superProductivity.exe"
+    },
```
Comment (Ash258) [inline bucket/super-productivity.json:20]:
```suggestion
```

File: bucket/super-productivity.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    "checkver": {
+        "github": "https://github.com/johannesjo/super-productivity/releases",
+        "regex": "v([\\d.]+)"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/johannesjo/super-productivity/releases/download/v$version/superProductivity-Setup-$version.exe#/dl.7z"
+            }
+        }
```
Comment (Ash258) [inline bucket/super-productivity.json:37]: Add hash extraction

File: bucket/super-productivity.json (hunk 3 above)
Comment (joao-neves95) [reply bucket/super-productivity.json]: I have fixed the update to support 64bit architecture only.

File: bucket/super-productivity.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                    "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-64.7z\" \"$dir\"",
+                    "Remove-Item \"$dir\\`$PLUGINSDIR\" -Force -Recurse"
+                ]
+            }
+        }
+    },
+    "uninstaller": {
+        "file": "Uninstall superProductivity.exe"
+    },
+    "bin": "superProductivity.exe",
```
Comment (Ash258) [inline bucket/super-productivity.json:21]: What are avialable parameters?

File: bucket/super-productivity.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    "shortcuts": [
+        [
+            "superProductivity.exe",
+            "superProductivity"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/johannesjo/super-productivity/releases",
+        "regex": "v([\\d.]+)"
+    },
```
Comment (Ash258) [inline bucket/super-productivity.json:31]: github checkver is enough

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/super-productivity.json (hunk 3 above)
Comment (joao-neves95) [reply bucket/super-productivity.json]: I'm going to need some help on the hash extraction @Ash258 , can you give me some guidance or examples?

Comment (Ash258) [reply bucket/super-productivity.json]: Stop commenting on resolved conversations...

Search in other manifests for latest.yml

File: bucket/super-productivity.json (hunk 5 above)
Comment (joao-neves95) [reply bucket/super-productivity.json:20]: Should I remove the uninstaller property?

File: bucket/super-productivity.json (hunk 6 above)
Comment (joao-neves95) [reply bucket/super-productivity.json:37]: I am analysing the hash extraction.

File: bucket/super-productivity.json (hunk 7 above)
Comment (joao-neves95) [reply bucket/super-productivity.json:21]: I don't think there are any input parameters.

Comment (Ash258) [reply bucket/super-productivity.json:21]: Then 

```suggestion
```

File: bucket/super-productivity.json (hunk 8 above)
Comment (joao-neves95) [reply bucket/super-productivity.json:31]: Should I remove the regex property?

File: bucket/super-productivity.json (hunk 7 above)
Comment (joao-neves95) [reply bucket/super-productivity.json:21]: Got it

File: bucket/super-productivity.json (hunk 8 above)
Comment (Ash258) [reply bucket/super-productivity.json:31]:
```suggestion
    "checkver": {
        "github": "https://github.com/johannesjo/super-productivity"
    },
```

File: bucket/super-productivity.json (hunk 7 above)
Comment (joao-neves95) [reply bucket/super-productivity.json:21]: If I remove the bin prop I receive the error "The term 'superproductivity' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a
path was included, verify that the path is correct and try again."

Comment (Ash258) [reply bucket/super-productivity.json:21]: That is the point. There is not a single reason to start the gui applications from command line, when there are no parameters....

File: bucket/super-productivity.json (hunk 5 above)
Comment (joao-neves95) [reply bucket/super-productivity.json:20]: Removing the uninstaller property does not bring any issues.
Will be committed.

File: bucket/super-productivity.json (hunk 4 above)
Comment (joao-neves95) [reply bucket/super-productivity.json]: Done

**Comment by joao-neves95** [comment]
Cool, thanks

**Comment by Ash258** [comment]
Can you stop spamming with irrelevant merge commits every 15 minutes??

**Comment by joao-neves95** [comment]
I was just keeping my branch updated, you can turn notifications off if you want.
If you want any more changes say.
Regards

**Review by Ash258 — APPROVED** [review]

**Comment by joao-neves95** [comment]
Hi,

There's a problem with the hash check:

```powershell
Installing 'super-productivity' (6.3.1) [64bit]
superProductivity-Setup-6.3.1.exe (53.4 MB) [===============================================================================================================================] 100%
Checking hash of superProductivity-Setup-6.3.1.exe ... ERROR Hash check failed!
App:         extras/super-productivity
URL:         https://github.com/johannesjo/super-productivity/releases/download/v6.3.1/superProductivity-Setup-6.3.1.exe#/dl.7z
First bytes: 4D 5A 90 00 03 00 00 00
Expected:    af4ff1d124d759220aadcce45452db475ead6ccb967c5f4b727ef516efff64b2ef22cc39ce679d566b013bfcb44739e61b0072005162c46c4bd91f145d40ebfd
Actual:      02ea79d6022eb1d1072ff647f33367ce87c425d065d9fe8654897a1d7bb92ae2560f3c2ca22de09c9df3fe5a144fe522b03ef23b7469fd3d95ebd49c4350459c

Please try again or create a new issue by using the following link and paste your console output:
https://github.com/lukesampson/scoop-extras/issues/new?title=super-productivity%406.3.1%3a+hash+check+failed
```

**Pull request merged**